### ISSUE TITLE
怎么训练完，效果还不如默认的模型，😓，大佬方便看下有什么问题吗？

### ISSUE THREAD

**Issue by 564142183** [PaddlePaddle/PaddleOCR#12136]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：docker，registry.baidubce.com/paddlepaddle/paddle:2.6.1-gpu-cuda12.0-cudnn8.9-trt8.6
- 版本号/Version：Paddle：2.6.1  PaddleOCR：2.7.5 问题相关组件/Related components：
- 运行指令/Command Code：
- 完整报错/Complete Error Message：
- 两次的识别结果
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/27718340/4795b233-82d2-4636-b988-4ddaacc87eff)

稍等附上训练过程

**Comment by 564142183** [comment]
数据集：https://aistudio.baidu.com/datasetdetail/107509
用PPOCRLabel标记了24张图片，如下
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/27718340/9f1e5c04-eca5-4890-8e6e-f627bcec2bcc)

### 1.划分数据集
`python PPOCRLabel/gen_ocr_train_val_test.py --trainValTestRatio 6:2:2 --datasetRootPath ./train_data --detRootPath ./train_data/det --recRootPath ./train_data/rec`
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/27718340/f38f08a3-ef82-4dc7-a9e2-8b1f79fb9d8c)


### 2.训练det模型
`python tools/train.py -c configs/det/ch_PP-OCRv4/ch_PP-OCRv4_det_cml.yml`
<details><summary>ch_PP-OCRv4_det_cml.yml</summary>
<p>
Global:
  debug: false
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 20
  save_model_dir: ./output/ch_PP-OCRv4_det
  save_epoch_step: 50
  eval_batch_step:
  - 0
  - 1000
  cal_metric_during_train: false
  checkpoints: null
  pretrained_model: ./train/model/ch_PP-OCRv4_det_train/best_accuracy.pdparams
  save_inference_dir: null
  use_visualdl: false
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./checkpoints/det_db/predicts_db.txt
  distributed: true
Architecture:
  name: DistillationModel
  algorithm: Distillation
  model_type: det
  Models:
    Student:
      model_type: det
      algorithm: DB
      Transform: null
      Backbone:
        name: PPLCNetV3
        scale: 0.75
        pretrained: false
        det: true
      Neck:
        name: RSEFPN
        out_channels: 96
        shortcut: true
      Head:
        name: DBHead
        k: 50
    Student2:
      pretrained: null
      model_type: det
      algorithm: DB
      Transform: null
      Backbone:
        name: PPLCNetV3
        scale: 0.75
        pretrained: true
        det: true
      Neck:
        name: RSEFPN
        out_channels: 96
        shortcut: true
      Head:
        name: DBHead
        k: 50
    Teacher:
      pretrained: https://paddleocr.bj.bcebos.com/PP-OCRv4/chinese/ch_PP-OCRv4_det_cml_teacher_pretrained/teacher.pdparams
      freeze_params: true
      return_all_feats: false
      model_type: det
      algorithm: DB
      Backbone:
        name: ResNet_vd
        in_channels: 3
        layers: 50
      Neck:
        name: LKPAN
        out_channels: 256
      Head:
        name: DBHead
        kernel_list:
        - 7
        - 2
        - 2
        k: 50
Loss:
  name: CombinedLoss
  loss_config_list:
  - DistillationDilaDBLoss:
      weight: 1.0
      model_name_pairs:
      - - Student
        - Teacher
      - - Student2
        - Teacher
      key: maps
      balance_loss: true
      main_loss_type: DiceLoss
      alpha: 5
      beta: 10
      ohem_ratio: 3
  - DistillationDMLLoss:
      model_name_pairs:
      - Student
      - Student2
      maps_name: thrink_maps
      weight: 1.0
      key: maps
  - DistillationDBLoss:
      weight: 1.0
      model_name_list:
      - Student
      - Student2
      balance_loss: true
      main_loss_type: DiceLoss
      alpha: 5
      beta: 10
      ohem_ratio: 3
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 2
  regularizer:
    name: L2
    factor: 5.0e-05
PostProcess:
  name: DistillationDBPostProcess
  model_name:
  - Student
  key: head_out
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5
Metric:
  name: DistillationMetric
  base_metric_name: DetMetric
  main_indicator: hmean
  key: Student
Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/
    label_file_list:
      - ./train_data/det/train.txt
    ratio_list: [1.0]
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - IaaAugment:
        augmenter_args:
        - type: Fliplr
          args:
            p: 0.5
        - type: Affine
          args:
            rotate:
            - -10
            - 10
        - type: Resize
          args:
            size:
            - 0.5
            - 3
    - EastRandomCropData:
        size:
        - 640
        - 640
        max_tries: 50
        keep_ratio: true
    - MakeBorderMap:
        shrink_ratio: 0.4
        thresh_min: 0.3
        thresh_max: 0.7
        total_epoch: 500
    - MakeShrinkMap:
        shrink_ratio: 0.4
        min_text_size: 8
        total_epoch: 500
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.485
        - 0.456
        - 0.406
        std:
        - 0.229
        - 0.224
        - 0.225
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - threshold_map
        - threshold_mask
        - shrink_map
        - shrink_mask
  loader:
    shuffle: true
    drop_last: false
    # batch_size_per_card: 16
    batch_size_per_card: 10
    num_workers: 8
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/
    label_file_list:
      - ./train_data/det/val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - DetLabelEncode: null
    - DetResizeForTest: 
        limit_side_len: 960
        limit_type: max
    - NormalizeImage:
        scale: 1./255.
        mean:
        - 0.485
        - 0.456
        - 0.406
        std:
        - 0.229
        - 0.224
        - 0.225
        order: hwc
    - ToCHWImage: null
    - KeepKeys:
        keep_keys:
        - image
        - shape
        - polys
        - ignore_tags
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 1
    num_workers: 2
profiler_options: null
</p>
</details> 

### 3.训练rec模型
`python tools/train.py -c configs/rec/PP-OCRv4/ch_PP-OCRv4_rec.yml`
<details><summary>ch_PP-OCRv4_rec.yml</summary>
<p>
Global:
  debug: false
  use_gpu: true
  epoch_num: 200
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/ch_PP-OCRv4_rec
  save_epoch_step: 10
  eval_batch_step: [0, 2000]
  cal_metric_during_train: true
  pretrained_model: ./train/model/ch_PP-OCRv4_rec_train/student.pdparams
  checkpoints:
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ppocr/utils/ppocr_keys_v1.txt
  max_text_length: &max_text_length 25
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv3.txt


Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
    warmup_epoch: 5
  regularizer:
    name: L2
    factor: 3.0e-05


Architecture:
  model_type: rec
  algorithm: SVTR_LCNet
  Transform:
  Backbone:
    name: PPLCNetV3
    scale: 0.95
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 120
            depth: 2
            hidden_dims: 120
            kernel_size: [1, 3]
            use_guide: True
          Head:
            fc_decay: 0.00001
      - NRTRHead:
          nrtr_dim: 384
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - NRTRLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: MultiScaleDataSet
    ds_width: false
    data_dir: ./train_data/
    ext_op_transform_idx: 1
    label_file_list:
    - ./train_data/rec/train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecConAug:
        prob: 0.5
        ext_data_num: 2
        image_shape: [48, 320, 3]
        max_text_length: *max_text_length
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  sampler:
    name: MultiScaleSampler
    scales: [[320, 32], [320, 48], [320, 64]]
    first_bs: &bs 192
    fix_bs: false
    divided_factor: [8, 16] # w, h
    is_training: True
  loader:
    shuffle: true
    batch_size_per_card: *bs
    drop_last: true
    num_workers: 8
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data
    label_file_list:
    - ./train_data/rec/val.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 128
    num_workers: 4
</p>
</details> 

### 4.导出推理模型
`python tools/export_model.py -c configs/det/ch_PP-OCRv4/ch_PP-OCRv4_det_cml.yml -o Global.pretrained_model=output/ch_PP-OCRv4_det/best_model/model.pdparams Global.save_inference_dir=inference_model/det/`
`python tools/export_model.py -c configs/rec/PP-OCRv4/ch_PP-OCRv4_rec.yml -o Global.pretrained_model="./output/ch_PP-OCRv4_rec/latest.pdparams" Global.save_inference_dir="./inference_model/rec/"`

### 5.执行脚本
```python
import cv2
  
from paddleocr import PaddleOCR

paddleocr = PaddleOCR(lang='ch', show_log=False, enable_mkldnn=True,
                      det_model_dir='./inference_model/det/Teacher',
                      rec_model_dir='./inference_model/rec'
)
img = cv2.imread('./train_data/0000.jpg')
result = paddleocr.ocr(img)
for i in range(len(result[0])):
    print(result[0][i][1][0])
```

**Comment by 564142183** [comment]
代码已上传到aistudio，https://aistudio.baidu.com/projectdetail/7900653?sUid=3018256&shared=1&ts=1716002266405

**Comment by UserWangZz** [comment]
数据量太少了

**Comment by xiaowen0407** [comment]
请问你也有出现这中情况吗  
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.ins_conv.1.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.ins_conv.2.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.ins_conv.3.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.inp_conv.0.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.inp_conv.1.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.inp_conv.2.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params neck.inp_conv.3.in_conv.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv1.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn1.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn1.bias not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn1._mean not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn1._variance not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv2.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv2.bias not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn2.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn2.bias not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn2._mean not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv_bn2._variance not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv3.weight not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.binarize.conv3.bias not in model
[2024/07/04 09:52:18] ppocr WARNING: The pretrained params head.thresh.conv1.weight not in model




用配置文件ch_PP-OCRv4_det_cml.yml   模型是官方的ch_PP-OCRv4_det_train/best_accuracy.pdparams

**Comment by 564142183** [comment]
没遇到过

**Comment by Sensorjang** [comment]
> 请问你也有出现这中情况吗 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.ins_conv.1.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.ins_conv.2.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.ins_conv.3.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.inp_conv.0.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.inp_conv.1.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.inp_conv.2.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 neck.inp_conv.3.in_conv.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv1.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn1.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn1.bias 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn1._mean 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn1._variance 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn1._variance 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv2.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv2.bias 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn2.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn2.bias 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn2._mean 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv_bn2._variance 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数head.binarize.conv3.weight 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.binarize.conv3.bias 不在模型中 [2024/07/04 09:52:18] ppocr 警告：预训练参数 head.thresh.conv1.weight 不在模型中
> 
> 使用配置文件ch_PP-OCRv4_det_cml.yml模型是官方的ch_PP-OCRv4_det_train/best_accuracy.pdparams

遇到了这个问题，目前不知道原因
